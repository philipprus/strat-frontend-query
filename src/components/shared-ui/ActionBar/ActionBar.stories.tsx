// Button.stories.ts | Button.stories.tsx

import { Meta } from "@storybook/react";
import { Button } from "reactstrap";
import Icon from "../Icon/Icon";
import ActionBar, { ActionBarProps } from "./ActionBar";

export default {
  component: ActionBar,
  title: "Shared-UI/ActionBar",
} as Meta;

export const Primary: React.VFC<ActionBarProps> = () => (
  <ActionBar
    title={`Current printing job  `}
    extra={
      <Button onClick={()=>{}} className="btn-icon btn-pause">
        <Icon icon="pause" width={13} height={14} />
      </Button>
    }
    progressValue={40}
  />
);
