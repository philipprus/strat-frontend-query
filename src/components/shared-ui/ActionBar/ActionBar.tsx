import React from "react";
import { Progress } from "reactstrap";

export type ActionBarProps = {
  title: JSX.Element | string;
  extra?: JSX.Element | string;
  progressValue?: number;
};
const ActionBar = ({ title, extra, progressValue }: ActionBarProps) => {
  return (
    <div className="actionbar">
      <div className="actionbar__title">{title}</div>
      <div className="actionbar__extra">{extra && extra}</div>
      {progressValue ? (
        <div className="actionbar__progress">
          <Progress value={progressValue} />
        </div>
      ) : null}
    </div>
  );
};

export default React.memo(ActionBar);
