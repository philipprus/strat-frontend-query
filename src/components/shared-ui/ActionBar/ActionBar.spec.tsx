import React from "react";
import renderer from "react-test-renderer";
import ActionBar from "./ActionBar";
import { shallow } from '../../../enzyme';

test("Link changes the class when hovered", () => {
  const component = renderer.create(<ActionBar title="Current printing job" />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("should show correct text", () => {
  const wrapper = shallow(<ActionBar title="Current printing job" />);
  expect(wrapper.text().includes("Current printing job")).toBe(true);
});
