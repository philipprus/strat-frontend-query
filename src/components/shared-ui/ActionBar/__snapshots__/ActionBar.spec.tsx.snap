// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<div
  className="actionbar"
>
  <div
    className="actionbar__title"
  >
    Current printing job
  </div>
  <div
    className="actionbar__extra"
  />
</div>
`;
